File: go/src/findtwothatsum/find_two_that_sum.go
package findtwothatsum

// FindTwoThatSum :
func FindTwoThatSum(numbers []int, sum int) (int, int) {
	for i, first := range numbers {
		for j, second := range numbers {
			if j <= i {
				continue
			}
			if first+second == sum {
				return i, j
			}
		}
	}
	return -1, -1
}

File: elixir/lib/polyglot/find_two_that_sum.ex
defmodule Polyglot.FindTwoThatSum do
  def iterate(numbers, sum) do
    res =
      for {first, i} <- Enum.with_index(numbers),
          {second, j} <- Enum.with_index(numbers),
          i < j,
          first + second == sum,
          do: {i, j}

    case List.first(res) do
      nil -> {-1, -1}
      x -> x
    end
  end
end
